eclass/ecm-common.eclass

```diff
@@ -0,0 +1,355 @@
+# Otherwise, add kde-frameworks/ki18n:* to BDEPEND, find KF[56]I18n and let
+# ki18n_install(po) generate and install translations.
+: "${ECM_I18N:=true}"
+
+# @ECLASS_VARIABLE: ECM_HANDBOOK
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Will accept "true" or "false" (default).  If set to "false", do nothing.
+# Otherwise, add "+handbook" to IUSE, add kde-frameworks/kdoctools:* to BDEPEND
+# find KF[56]DocTools in CMake, call add_subdirectory(ECM_HANDBOOK_DIRS)
+# and let let kdoctools_install(po) generate and install translated docbook
+# files.
+: "${ECM_HANDBOOK:=false}"
+
+# @ECLASS_VARIABLE: ECM_HANDBOOK_DIRS
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Default is "doc" which is correct for the vast majority of packages. Specifies
+# one or more directories containing untranslated docbook file(s) relative to
+# ${S} to be added via add_subdirectory.
+if [[ ${ECM_HANDBOOK_DIRS} ]]; then
+	[[ ${ECM_HANDBOOK_DIRS@a} == *a* ]] ||
+		die "ECM_HANDBOOK_DIRS must be an array"
+else
+	ECM_HANDBOOK_DIRS=( doc )
+fi
+
+# @ECLASS_VARIABLE: ECM_INSTALL_FILES
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Array of <file>:<destination_path> tuples to install by CMake via
+# install(FILES <file> DESTINATION <destination_path>)
+if [[ ${ECM_INSTALL_FILES} ]]; then
+	[[ ${ECM_INSTALL_FILES@a} == *a* ]] ||
+		die "ECM_INSTALL_FILES must be an array"
+fi
+
+# @ECLASS_VARIABLE: ECM_INSTALL_ICONS
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Array of <icon>:<icon_install_dir> tuples to feed to ECMInstallIcons
+# via ecm_install_icons(ICONS <icon> DESTINATION <icon_install_dir)
+if [[ ${ECM_INSTALL_ICONS} ]]; then
+	[[ ${ECM_INSTALL_ICONS@a} == *a* ]] ||
+		die "ECM_INSTALL_ICONS must be an array"
+fi
+
+# @ECLASS_VARIABLE: ECM_KCM_TARGETS
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Array of <target>:<subdir> tuples to feed to ECMInstallIcons via
+# ecmcommon_generate_desktop_file(<target> <subdir>), which is this
+# eclass adaptation of kcmutils_generate_desktop_file.
+if [[ ${ECM_KCM_TARGETS} ]]; then
+	[[ ${ECM_KCM_TARGETS@a} == *a* ]] ||
+		die "ECM_KCM_TARGETS must be an array"
+fi
+
+DESCRIPTION="Common files for ${PN/-common/}"
+
+BDEPEND=">=kde-frameworks/extra-cmake-modules-5.115.0:*"
+
+case ${ECM_I18N} in
+	true)
+		KF5_BDEPEND+=( "kde-frameworks/ki18n:5" )
+		KF6_BDEPEND+=( "kde-frameworks/ki18n:6" )
+		;;
+	false) ;;
+	*)
+		eerror "Unknown value for \${ECM_I18N}"
+		die "Value ${ECM_I18N} is not supported"
+		;;
+esac
+
+case ${ECM_HANDBOOK} in
+	true)
+		IUSE+=" +handbook"
+		KF5_BDEPEND+=( "handbook? ( kde-frameworks/kdoctools:5 )" )
+		KF6_BDEPEND+=( "handbook? ( kde-frameworks/kdoctools:6 )" )
+		;;
+	false) ;;
+	*)
+		eerror "Unknown value for \${ECM_HANDBOOK}"
+		die "Value ${ECM_HANDBOOK} is not supported"
+		;;
+esac
+
+if [[ ${ECM_KCM_TARGETS} ]]; then
+	KF5_BDEPEND+=( "kde-frameworks/kcmutils:5" )
+	KF6_BDEPEND+=( "kde-frameworks/kcmutils:6" )
+fi
+
+if [[ ${KF6_BDEPEND} && ${KF5_BDEPEND} ]]; then
```

Comment:
We could tidy this up with an array of vars then `${!X}` but I don't think it's worth it for this.